### PR TITLE
chore: reduce CI time

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

- [x] Update CI Trigger policy that run CI conditionally

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: .github/workflows/markdown-link-checker.yml [hunk 1]
```diff
@@ -1,14 +0,0 @@
-name: Check markdown links
```
Comment (nic-chen) [inline .github/workflows/markdown-link-checker.yml:1]: we don't need this CI now ?

Comment (juzhiyuan) [reply .github/workflows/markdown-link-checker.yml:1]: yes, because Apache Infra doen't allow this one

Comment (juzhiyuan) [reply .github/workflows/markdown-link-checker.yml:1]: See this one: https://cwiki.apache.org/confluence/display/BUILDS/GitHub+Actions+status#GitHubActionsstatus-Security

**Review by starsz — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Comment by juzhiyuan** [comment]
wait for #1761

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**